src/benchcab/fluxsite.py

```diff
@@ -212,9 +227,11 @@ def run(self):
         self.logger.debug(f"Running task {task_name}... CABLE standard output ")
         self.logger.debug(f"saved in {task_dir / internal.CABLE_STDOUT_FILENAME}")
 
+        self._state_init()
         try:
             self.run_cable()
             self.add_provenance_info()
+            self._state_set("done")
```

Comment:
In fact, it might be nice to have a state class?

```python
class State:

    def set(self, state):
        # ... get the dir, set the state

    def get(self):
        # ... get the most recent state file

    def done(self):
        self.set('done')

    def is_done(self):
        return self.get() == 'done'
```

....or similar

This will make interacting with the state framework more concise.

Reply:
```python
    def get(self):
        # ... get the most recent state file
```

Is there a use case to retrieve the most recent state file?

Reply:
Depending on how you implement this, it might be useful for any master script to be able to get the state of the child processes it spawned - you would need to allow arguments to find the correct dir/file though. Up to you, I just usually put a getter with a setter whenever developing classes as they often come in handy.

Reply:
Sorry, I was asking specifically about the "most recent" requirement. I've gone and implemented the `get` method anyway but I'm not currently using the functionality:

```python
    def get(self) -> str:
        """Get the state of the most recent state attribute."""
        attrs = list(self.state_dir.glob(f"{STATE_PREFIX}*"))

        def get_mtime(path: Path):
            return path.stat().st_mtime

        attrs.sort(key=get_mtime)
        try:
            return attrs.pop().name.removeprefix(STATE_PREFIX)
        except IndexError as exc:
            msg = "No attributes have been set."
            raise StateAttributeError(msg) from exc
```